src/collector.rs

```diff
@@ -154,7 +154,7 @@ pub(crate) async fn init_future(
     Ok(())
 }
 
-pub(crate) fn new(config: Config) -> (RecorderHandle, impl Future<Output = ()>) {
```

Comment:
Why was this necessary?

Reply:
It was used in benchmarks

Reply:
I don't like that we expose it in the public API :/